src/main/java/org/breedinginsight/brapps/importer/services/processors/ExperimentProcessor.java

```diff
@@ -1681,18 +2209,49 @@ private Set<String> fetchTrialDbidsForStudies(Set<String> studyDbIds, Program pr
         return trialDbIds;
     }
 
-    private void processAndCacheTrial(BrAPITrial existingTrial, Program program, String trialRefSource, Map<String, PendingImportObject<BrAPITrial>> trialByNameNoScope) {
+    private void processAndCacheTrial(
+        BrAPITrial existingTrial, 
+        Program program,
+        Map<String, PendingImportObject<BrAPITrial>> trialByNameNoScope) {
 
         //get TrialId from existingTrial
-        BrAPIExternalReference experimentIDRef = Utilities.getExternalReference(existingTrial.getExternalReferences(), trialRefSource)
+        BrAPIExternalReference experimentIDRef = Utilities.getExternalReference(existingTrial.getExternalReferences(), String.format("%s/%s", BRAPI_REFERENCE_SOURCE, ExternalReferenceSource.TRIALS.getName()))
                                                           .orElseThrow(() -> new InternalServerException("An Experiment ID was not found in any of the external references"));
-        UUID experimentId = UUID.fromString(experimentIDRef.getReferenceID());
+        UUID experimentId = UUID.fromString(experimentIDRef.getReferenceId());
 
         trialByNameNoScope.put(
                 Utilities.removeProgramKey(existingTrial.getTrialName(), program.getKey()),
                 new PendingImportObject<>(ImportObjectState.EXISTING, existingTrial, experimentId));
     }
 
+    /**
+     * This function collates unique ObsUnitID values from a list of BrAPIImport objects.
+     * It iterates through the list and adds non-blank ObsUnitID values to a Set. It also checks for any repeated ObsUnitIDs.
+     *
+     * @param importRows a List of BrAPIImport objects containing ExperimentObservation data
+     * @return a Set of unique ObsUnitID strings
+     * @throws IllegalStateException if a repeated ObsUnitID is encountered
+     */
+    private Set<String> collateReferenceOUIds(List<BrAPIImport> importRows) {
```

Comment:
You might want to indicate that the `hasAllReferenceUnitIds` and `hasNoReferenceUnitIds` instance variables are set/mutated as well.